src/main/java/soconnect/logic/commands/SearchCommand.java

```diff
@@ -20,26 +22,32 @@ public class SearchCommand extends Command {
     public static final String EMPTY_CONDITION = "";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Searches all persons whose information contains "
+            + ": Searches all people whose information contains "
             + "the specified keyword (case-insensitive) and displays them as a list with index numbers.\n"
             + "Parameters: KEYWORD [CONDITION] [KEYWORD]...\n"
             + "Example: " + COMMAND_WORD + " t/friend, "
-            + COMMAND_WORD + " " + AND_CONDITION + " n/John a/NUS";
+            + COMMAND_WORD + " " + AND_CONDITION + " n/John a/NUS"
+            + COMMAND_WORD + " " + OR_CONDITION + " p/12345678 e/betsy@nus.edu";
 
     private final Predicate<Person> predicate;
+    private final Predicate<Person> alternativePredicate;
 
     /**
      * Constructs a {@code SearchCommand} to search contacts in SoConnect.
      */
-    public SearchCommand(Predicate<Person> predicate) {
-        requireNonNull(predicate);
+    public SearchCommand(Predicate<Person> predicate, Predicate<Person> alternativePredicate) {
+        requireAllNonNull(predicate, alternativePredicate);
         this.predicate = predicate;
+        this.alternativePredicate = alternativePredicate;
     }
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         model.updateFilteredPersonList(predicate);
+        if (model.getFilteredPersonList().size() == 0) {
```

Comment:
One possible suggestion for refactoring (can be ignored since its very minor), implement a method in model to check for empty list, so that the condition can become something like `model.isFilteredPersonListEmpty()`